index.js

```diff
@@ -78,6 +79,7 @@ export default class Carousel extends Component {
         currentPage: props.currentPage,
         size,
         childrenLength,
+        contents: null,
```

Comment:
I moved this into state as a change in `contents` will now be the trigger that forces the final render.